staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -79,7 +81,15 @@ func getCallers(r interface{}) string {
 
 // ErrorHandlers is a list of functions which will be invoked when an unreturnable
 // error occurs.
-var ErrorHandlers = []func(error){logError}
+// TODO(lavalamp): for testability, this and the below HandleError function
+// should be packaged up into a testable and reusable object.
+var ErrorHandlers = []func(error){
+	logError,
+	(&rudimentaryErrorBackoff{
```

Comment:
Ideally, I would like to see us formalizing error handling in the controller framework. Error handling differs between controllers and this is indeed a change that affects all controllers which is not good. Most of them already use a backoff mechanism as @deads2k mentioned.